src/WebJobs.Extensions.OpenAI/Search/ISearchProvider.cs

```diff
@@ -31,7 +31,7 @@ public record SearchableDocument(
     public ConnectionInfo? ConnectionInfo { get; set; }
 }
 
-public record ConnectionInfo(string ConnectionName, string CollectionName);
+public record ConnectionInfo(string ConnectionName, string CollectionName, string? ApiKey);
```

Comment:
Consider `Credentials` instead of `ApiKey` since I suspect that some providers won't have an "API Key" concept. "Credentials" is a more generic name that would cover more scenarios.

Reply:
used `CredentialSettingName` as per another comment